### PR TITLE
luarocks-bootstrap: cmake and zip should be propagatedNativeBuildInputs

### PR DESCRIPTION
**Pull request by Mic92**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mic92** [comment]
Probably should go into staging...

**Review by 7c6f434c — APPROVED** [review]
Looks good. But yeah, maybe to staging

**Comment by teto** [comment]
I dont quite get the difference. propagatedNativeBuildInputs

**Comment by Mic92** [comment]
@teto the issue just now is that cmake is build for arm64. But I am also wondering why does this need to be a propagating build input? If a build system is required for an ecosystem than either a builder or a build hook should be used. Propagating pulls in cmake in places where they are not required. This is different from python libraries.

File: pkgs/development/tools/misc/luarocks/luarocks-nix.nix [hunk 1]
```diff
@@ -16,7 +16,7 @@ luarocks_bootstrap.overrideAttrs (old: {
     hash = "sha256-hsjv+jlLsoIDM4gB/0mFeoVu1YZ1I9ELDALLTEnlCF0=";
   };
 
-  propagatedBuildInputs = old.propagatedBuildInputs ++ [
+  propagatedNativeBuildInputs = old.propagatedNativeBuildInputs ++ [
```
Comment (mrcjkb) [inline pkgs/development/tools/misc/luarocks/luarocks-nix.nix:19]: question: are you sure `file` and `nurl` should be `propagatedNativeBuildInputs` too?

**Comment by 7c6f434c** [comment]
@teto intuitively, builds depending on a cross build are also cross builds, so they need the native to the build machine `cmake`, no?

**Comment by teto** [comment]
> the issue just now is that cmake is build for arm64 when cross compiling from an x86 machine. But I am also wondering why does this need to be a propagating build input? If a build system is required for an ecosystem than either a builder or a build hook should be used

luarocks accepts "cmake" as a build format https://github.com/luarocks/luarocks/wiki/Rockspec-format#cmake so luarocks might decide to run cmake at runtime. Asking the luarocks maintainer, he thought this was actually not a good idea/advised and I think we only had one package (luv) which used to relied on it and then we worked around that because it was not practical. So if that helps you/we could drop cmake from the propagatedNativeBuildInputs

File: pkgs/development/tools/misc/luarocks/luarocks-nix.nix (hunk 1 above)
Comment (teto) [reply pkgs/development/tools/misc/luarocks/luarocks-nix.nix:19]: they should just be in the wrapper. proabaly for me to fix though.

**Review by teto — APPROVED** [review]

**Comment by teto** [comment]
we probably dont need to propagate zip either, which is used to create `*.rock` files. (a .zip with the extension renamed)

**Comment by 7c6f434c** [comment]
Merging this as an improvement, some things can be made non-propagated or moved into wrappers in follow-ups, I guess.

**Pull request merged**